File: JANUARY/16-01-2026/Minimum Number of Workers.js
class Solution {
    minMen(arr) {
        const n = arr.length;
        const intervals = [];
        
        // Build intervals
        for (let i = 0; i < n; i++) {
            if (arr[i] !== -1) {
                const start = Math.max(0, i - arr[i]);
                const end = Math.min(n - 1, i + arr[i]);
                intervals.push([start, end]);
            }
        }
        
        if (intervals.length === 0) return -1;
        
        // Sort intervals by start
        intervals.sort((a, b) => a[0] - b[0]);
        
        let ans = 0;
        let i = 0;
        let covered = 0;
        let farthest = -1;
        
        while (covered < n) {
            let found = false;
            while (i < intervals.length && intervals[i][0] <= covered) {
                farthest = Math.max(farthest, intervals[i][1]);
                i++;
                found = true;
            }
            if (!found) return -1; // gap found
            covered = farthest + 1;
            ans++;
        }
        
        return ans;
    }
}

File: JANUARY/16-01-2026/Minimum Number of Workers.py
class Solution:
    def minMen(self, arr):
        n = len(arr)
        intervals = []
        
        # Build intervals for each available person
        for i, val in enumerate(arr):
            if val != -1:
                start = max(0, i - val)
                end = min(n - 1, i + val)
                intervals.append((start, end))
        
        if not intervals:
            return -1
        
        # Sort intervals by start
        intervals.sort()
        
        ans = 0
        i = 0
        covered = 0
        farthest = -1
        
        while covered < n:
            found = False
            while i < len(intervals) and intervals[i][0] <= covered:
                farthest = max(farthest, intervals[i][1])
                i += 1
                found = True
            if not found:
                return -1  # gap found
            covered = farthest + 1
            ans += 1
        
        return ans